File: camerax_android/lib/src/core/camera_selector.dart
// import 'package:camerax_android/src/jni.dart' as $native;
// import 'package:camerax_platform_interface/camerax_platform_interface.dart'
//     as $interface;

// final class CameraSelector extends $interface.CameraSelector {
//   final $native.CameraSelector obj;

//   CameraSelector.$native(this.obj) : super.impl();

//   factory CameraSelector({
//     $interface.LensFacing? lensFacing,
//   }) {
//     final builderObj = $native.CameraSelector$Builder.new$1();
//     if (lensFacing != null) {
//       builderObj.requireLensFacing(lensFacing.obj);
//     }
//     final obj = builderObj.build();
//     return CameraSelector.$native(obj);
//   }

//   @override
//   int get hashCode => obj.hashCode;

//   @override
//   bool operator ==(Object other) => other is CameraSelector && obj == other.obj;
// }

// extension LensFacingArgs on $interface.LensFacing {
//   $native.CameraSelector$LensFacing get obj {
//     switch (this) {
//       case $interface.LensFacing.unknown:
//         return $native.CameraSelector$LensFacing.UNKNOWN!;
//       case $interface.LensFacing.front:
//         return $native.CameraSelector$LensFacing.FRONT!;
//       case $interface.LensFacing.back:
//         return $native.CameraSelector$LensFacing.BACK!;
//       case $interface.LensFacing.external:
//         return $native.CameraSelector$LensFacing.EXTERNAL!;
//     }
//   }
// }

File: camerax_android/android/src/main/kotlin/dev/hebei/camerax_android/legacy/core/CameraSelector.kt
package dev.hebei.camerax_android.legacy.core

import dev.hebei.camerax_android.legacy.CameraXRegistrar
import dev.hebei.camerax_android.legacy.LensFacing
import dev.hebei.camerax_android.legacy.PigeonApiCameraSelector

class CameraSelector(registrar: CameraXRegistrar) : PigeonApiCameraSelector(registrar) {
    override fun pigeon_defaultConstructor(lensFacing: LensFacing?): androidx.camera.core.CameraSelector {
        val builder = androidx.camera.core.CameraSelector.Builder()
        if (lensFacing != null) {
            builder.requireLensFacing(lensFacing.obj)
        }
        return builder.build()
    }

    override fun front(): androidx.camera.core.CameraSelector {
        return androidx.camera.core.CameraSelector.DEFAULT_FRONT_CAMERA
    }

    override fun back(): androidx.camera.core.CameraSelector {
        return androidx.camera.core.CameraSelector.DEFAULT_BACK_CAMERA
    }

    override fun external(): androidx.camera.core.CameraSelector {
        return androidx.camera.core.CameraSelector.Builder()
            .requireLensFacing(androidx.camera.core.CameraSelector.LENS_FACING_EXTERNAL).build()
    }
}

val LensFacing.obj
    get() = when (this) {
        LensFacing.UNKNOWN -> androidx.camera.core.CameraSelector.LENS_FACING_UNKNOWN
        LensFacing.FRONT -> androidx.camera.core.CameraSelector.LENS_FACING_FRONT
        LensFacing.BACK -> androidx.camera.core.CameraSelector.LENS_FACING_BACK
        LensFacing.EXTERNAL -> androidx.camera.core.CameraSelector.LENS_FACING_EXTERNAL
    }

val Int.lensFacingArgs
    get() = when (this) {
        androidx.camera.core.CameraSelector.LENS_FACING_UNKNOWN -> LensFacing.UNKNOWN
        androidx.camera.core.CameraSelector.LENS_FACING_FRONT -> LensFacing.FRONT
        androidx.camera.core.CameraSelector.LENS_FACING_BACK -> LensFacing.BACK
        androidx.camera.core.CameraSelector.LENS_FACING_EXTERNAL -> LensFacing.EXTERNAL
        else -> throw IllegalArgumentException()
    }